File: week-05/day-05/garden-again/flower.ts
import { Plant } from "./plant";

'use strict';

export class Flower extends Plant {

  constructor(color: string){
    super(color)
    this.absorbingCap = 0.75;
  }

  needWater(): boolean {
    return this.currentWaterAm<5;
  }  
  
  getType() {
    return 'flower'
  }
  
}

File: datascience/week-01/garden/garden_py/flower.py
import random
import sys
import os
from plant import Plants


class Flower(Plants):

    def __init__(self, color):
        super().__init__(color)
        self._absorbingCap = 0.75

    def needWater(self):
        return self._currentWaterAm < 5

    def get_type(self):
        return 'flower'
